packages/block-editor/src/hooks/variation.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, store as blocksStore } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+import { useContext, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	GlobalStylesContext,
+	toStyles,
+	getBlockSelectors,
+} from '../components/global-styles';
+import { useStyleOverride } from './utils';
+import { store as blockEditorStore } from '../store';
+
+function getVariationNameFromClass( className ) {
+	const match = className?.match( /\bis-style-(?!default)(\S+)\b/ );
+	return match ? match[ 1 ] : null;
+}
+
+function useBlockSyleVariation( name, variation, clientId ) {
+	const { user: userStyles } = useContext( GlobalStylesContext );
+	const { globalSettings, globalStyles } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return {
+			globalSettings: getSettings().__experimentalFeatures,
+			globalStyles: getSettings().__experimentalStyles,
+		};
```

Comment:
Completely ignorable.

```suggestion
		const { __experimentalFeatures, __experimentalStyles } =
			select( blockEditorStore ).getSettings();
		return {
			globalSettings: __experimentalFeatures,
			globalStyles: __experimentalStyles,
		};
```

Reply:
Anything that cleans up the code is fine by me! I think this is likely one of a few places like this that needs tidying up after multiple iterations, backflips, edge cases, and dead ends 😅